### DISCUSSION TITLE
日志配置不生效

### DISCUSSION THREAD

**Discussion by KingDracula** [midwayjs/midway#3879] [Q&A] [answered]
### Describe the problem（描述问题）
const processPath = process.cwd();
![image](https://github.com/midwayjs/midway/assets/13584613/1026747e-376e-4fbf-a726-2eb105eef052)
设置log输出目录是当前工程下了，但是通过断点看配置，是输出到了用户文件夹下了，然后控制台不打印ctx.logger的日志内容了，看输出的log文件里是有的，麻烦问一下这个要怎么设置

### Midway Versions（Midway 版本）
✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(not installed)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.16.0)
✓ @midwayjs/bull(not installed)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.16.0)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(3.16.0)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(not installed)
✓ @midwayjs/info(3.16.1)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.16.0)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(3.16.0)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(3.16.1)
✓ @midwayjs/session(not installed)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(3.16.1)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/tenant(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(not installed)
✓ @midwayjs/upload(3.16.1)
✓ @midwayjs/validate(3.16.1)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.16.1)
✓ @midwayjs/ws(not installed)

**Comment by KingDracula** [comment]
![image](https://github.com/midwayjs/midway/assets/13584613/95034baa-25bb-42f7-b37f-3dd791853a76)
改成这样日志输出目录正常了，但是 ctx.logger.info(` rt = ${Date.now() - startTime}ms`);  这个的log 控制台上还是没有打印

**Comment by KingDracula** [comment]
![image](https://github.com/midwayjs/midway/assets/13584613/cbc21393-1f71-4f6b-b398-fe8cd86d81be)
配置改成这样可以正常打印了，不知道什么原因把默认配置给覆盖了

**Reply by czy88840616** [reply]
因为你只改了 default，default 是会和其他 logger 合并的，优先级不高，看这里就能理解。https://github.com/midwayjs/logger/blob/main/src/factory.ts#L102

**Comment by KingDracula** [comment]
给console添加 format 属性报错，提示 Cannot read properties of undefined (reading 'format')

![image](https://github.com/midwayjs/midway/assets/13584613/9c38690b-b6d3-429e-a0fd-0189efba2cba)

![image](https://github.com/midwayjs/midway/assets/13584613/7013f339-d638-4a2f-bdcd-5b02f2079aa7)

**Reply by czy88840616** [reply, marked as answer]
fixed in @midwayjs/logger@3.4.2

**Reply by KingDracula** [reply]
升级后可以了，多谢了
